### DISCUSSION TITLE
How can I safely change a value in a collection?

### DISCUSSION THREAD

**Discussion by Hiwo-1101000-1101001** [pocketbase/pocketbase#5060] [Q&A]
Hi, I'm just getting involved in pocketbase and wanted to make a simple blog to understand how to work with it. I have a question how to implement the increment of views so that it is safe. I only came up with an option to remove the ban on updating posts from everyone and do something like:
`await pb.collection('posts').update(id, {views: response.views + 1});`

but this option, as I understand it, allows everyone to change the content of the post according to a certain url request, which is not very safe. Although this code example above is executed on the server side, in any case, as I understand it, I need to use something like middleware to determine the permission of the post change request by token. Or maybe you have another solution to this issue. I will be waiting for your answer, thank you!

**Comment by godmar** [comment]
Based on my understanding of PB so far, I believe you have two choices:

- add a hook that updates the number of views. You'll want to run this in a transaction to ensure the atomicity of the update (if you care about 100% accuracy).  
- run this with admin credentials from server side code and set the API rules to allow only admins to perform this update. Care may be required to set the rules correctly. If you make any updates admin-only then this may preclude you, for instance, from making updates to a `posts` record with the credential of, say the owner of the post.  So you may need a more complex expression in the updateRule: perhaps something that says if the update request includes a new value for the `views` field, then require admin, else check if the requesting authorized user id is equal to the post's owner, or similar.

I'm curious as to what the best practice for this problem/use case is.

**Comment by TobiasNickel** [comment]
I suggest to not just a count-field , but a log table that you can count afterwards once you find malicious view entries, you can just filter them. so, public would only get the create permission. you can make a rule to make it difficult/inconvenient inserting a wrong value, such as use a relation collumn to your post and check the time to be today(the current day)
